### ISSUE TITLE
Fix Pyret Game Starter File 

### ISSUE THREAD

**Issue by flannery-denny** [bootstrapworld/curriculum#1783]
@schanzer Looks like we had updated the Sam the Butterfly files but not the Game Starter File. I've made the changes that I can, but it still doesn't run because `update-player` is using `posn` under the hood in the 1.7 Context. 

Guessing we need a new context because it looks like 1.7 has already been in use? 

Anyhow, here's the new [Blank Game Starterfile](https://drive.google.com/file/d/14SBm-kWL2jdrOGOQXWv-30mFaU50hTqF/view?usp=drive_link) I just made for the 2024 folder. Lmk when the file is working. Thanks!

**Comment by schanzer** [comment]
@flannery-denny can you say more about what doesn't match? Or is in your head and you're just making the issue to track it, so the rest of us don't need to worry about?

**Comment by flannery-denny** [comment]
@schanzer Not sure whether you made that request of me as I was making the issue description? I've changed the issue title to reflect what's left to do. Lmk if the description isn't clear enough.

**Comment by schanzer** [comment]
@flannery-denny I'm still not sure what's wrong with the file. It's _supposed_ to use posn under the hood. It did in previous years as well. Are you sure this isn't just the same problem with the `constants` library?

**Comment by schanzer** [comment]
@flannery-denny okay. This one was NOT EASY, and required some hacks from Ben just to make it possible. 

I spent the evening rewriting most of the game-handling code, and we now have something that's a lot closer to WeScheme in functionality. Specifically, students can start their functions taking just `x` and producing the next `x`, and then later change that same function to consume `x` and `y` and produce a `posn`....all without having to load a different file or call a different `make-game` function.

I went ahead and rewrote Sam the Butterfly as well, so that `is-onscreen` can evolve without kids having to change anything. Now they can just call `sam` and `trace`, and everything will Just Work(tm).

This will probably allow us to simplify the student procedures quite a bit, not to mention get rid of a ton of `@ifproglangs` in a few lessons.
